src/twisted/conch/test/test_transport.py

```diff
@@ -310,36 +310,18 @@ def getPrimes(self):
         return {2048: (group14,), 4096: ((5, 7),)}
 
 
-class MockOldFactoryPublicKeys(MockFactory):
+class LegacyGetDHPrimesFactory(MockFactory):
     """
-    The old SSHFactory returned mappings from key names to strings from
-    getPublicKeys().  We return those here for testing.
+    A factory that implements L{twisted.conch.ssh.factory.SSHFactory.getDHPrime}
+    with a single argument.
     """
 
-    def getPublicKeys(self):
+    def getDHPrime(self, ideal):
```

Comment:
It would still be nice to have the type-hint here for completeness:

```suggestion
    def getDHPrime(self, ideal: int) -> Tuple[int, int]:
```